drivers/at/at.c

```diff
@@ -22,39 +23,79 @@
 #define AT_PRINT_INCOMING (0)
 #endif
 
-static void _isrpipe_write_one_wrapper(void *_isrpipe, uint8_t data)
+#if defined(MODULE_AT_URC_ISR_LOW)
+#define AT_EVENT_PRIO EVENT_PRIO_LOW
+#elif defined(MODULE_AT_URC_ISR_MEDIUM)
+#define AT_EVENT_PRIO EVENT_PRIO_MEDIUM
+#elif defined(MODULE_AT_URC_ISR_HIGH)
+#define AT_EVENT_PRIO EVENT_PRIO_HIGH
+#endif
+
+#if defined(MODULE_AT_URC_ISR)
+void _event_process_urc(event_t *_event)
 {
-    isrpipe_write_one(_isrpipe, data);
+    at_dev_t *dev = (at_dev_t *)container_of(_event, at_dev_t, event);
+    at_process_urc(dev, 1000);
+}
+#endif
+
+static void _isrpipe_write_one_wrapper(void *_dev, uint8_t data)
+{
+    at_dev_t *dev = (at_dev_t *) _dev;
+    isrpipe_write_one(&dev->isrpipe, data);
+#if defined(MODULE_AT_URC_ISR)
+    if (data == AT_RECV_EOL_2[0] && !dev->awaiting_response) {
+        event_post(AT_EVENT_PRIO, &dev->event);
+    }
+#endif
 }
 
 int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize)
 {
     dev->uart = uart;
+
+#if IS_USED(MODULE_AT_URC_ISR)
+    dev->awaiting_response = false;
+    dev->event.handler = _event_process_urc;
+#endif
+
     isrpipe_init(&dev->isrpipe, (uint8_t *)buf, bufsize);
 
     return uart_init(uart, baudrate, _isrpipe_write_one_wrapper,
-                     &dev->isrpipe);
+                     dev);
```

Comment:
Why do you need this change ? The old version seems still valid to me and now the isrpipe attribute is used implicitly.

Reply:
Ah, nevermind, I understand :)